### PR TITLE
For #318: Added Item history 

### PR DESCRIPTION
**Pull request by paulodamaso**
For #318:
- Implemented item_history method
- Implemented item assign method

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#325](https://github.com/timelesslounge/timelessis/issues/325) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#325](https://github.com/timelesslounge/timelessis/issues/325) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/36faa716-a91a-456c-8169-5adf4fad3ba7); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/test_items.py [hunk 1]
```diff
@@ -43,18 +46,48 @@ def test_new_item_history():
...
     """ Test item assign """
+
+    company = Company(
+        id=223,
+        name="Bad Company",
+        code="Bad Cmpny",
+        address="addr"
+    )
+
+    location = Location(
```
Comment (amihaiemil) [inline tests/test_items.py]: @paulodamaso It doesn't seem like you need to create a ``location`` for this test, or? As I see it's ``Company -> Item -> Employee``, location is irrelevant.

File: tests/test_items.py [hunk 2]
```diff
@@ -43,18 +46,48 @@ def test_new_item_history():
...
-        company_id=223
+        comment="Eggs from ducks",
+        company_id=company.id,
+        created_on=datetime.utcnow,
+        updated_on=datetime.utcnow,
+        company=company
     )
 
-    new_employee = Item(
+    new_employee = Employee(
```
Comment (amihaiemil) [inline tests/test_items.py]: @paulodamaso I would name this variable ``receiver`` or ``assignee`` (the one who receives or is assigned an Item)

File: tests/test_items.py [hunk 3]
```diff
@@ -43,18 +46,48 @@ def test_new_item_history():
...
+        latitude="44332211",
+        status="T",
+        company_id=223,
+        poster_id=100,
+        synchronized_on=datetime.utcnow,
+        working_hours=1,
+        closed_days=2
+    )
+
     new_item = Item(
```
Comment (amihaiemil) [inline tests/test_items.py]: @paulodamaso Just ``item`` (without ``new_``) should be enough, I think.

File: tests/test_items.py [hunk 4]
```diff
@@ -73,16 +106,42 @@ def test_item_assign():
...
-    first_employee = Item(
+
+    company = Company(
+        id=223,
+        name="Bad Company",
+        code="Bad Cmpny",
+        address="addr"
+    )
+    
+    location = Location(
```
Comment (amihaiemil) [inline tests/test_items.py]: @paulodamaso same here, location seems to be irrelevant for the test case

File: timeless/items/models.py [hunk 5]
```diff
@@ -29,16 +29,19 @@ def __init__(self, **kwargs):
     def assign(self, employee):
         """ Assigning the item to an employee """
         self.employee_id = employee.id
-        item_history = ItemHistory(employee_id=self.employee_id, item_id=self.id)
+        self.history.append(
+            ItemHistory(
+                employee_id=self.employee_id,
+                item_id=self.id,
+                start_time=datetime.utcnow
+            )
```
Comment (amihaiemil) [inline timeless/items/models.py]: @paulodamaso before simply appending the new ``ItemHistory``, maybe we can iterate over them, find the latest with ``item_id == self.id`` and set its ``end_time`` (that is, the time when the Item was handed over from one employee to the next. Makes sense?

**Comment by paulodamaso** [comment]
@amihaiemil Done, please take a look

**Comment by paulodamaso** [comment]
@amihaiemil Fixed!

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@emilianodellacasa Looks like the error wasn't related to my changes, I've merged the master branch, please take a look now

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#325](https://github.com/timelesslounge/timelessis/issues/325) is not in the agenda of @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), can't inspect

**Comment by 0crat** [comment]
The job [#325](https://github.com/timelesslounge/timelessis/issues/325) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by amihaiemil** [comment]
@ypshenychka review here? :D

**Comment by ypshenychka** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/325#issuecomment-466333494))

@ypshenychka This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#325](https://github.com/timelesslounge/timelessis/issues/325) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 50min
  * @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) will receive 15 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23325%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +20 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)
